inc/namespace.php

```diff
@@ -56,17 +56,73 @@ function get_merged_config() : array {
 
 	// Get custom config overrides.
 	$composer_file = ROOT_DIR . '/composer.json';
+
 	$composer_json = get_json_file_contents_as_array( $composer_file );
+	$environment = get_environment_type();
 
 	$config = merge_config_settings( $default_config, $composer_json['extra']['altis'] ?? [] );
 
+	validate_config_settings( $config );
 	// Look for environment specific settings in the config and merge it in.
-	$environment = get_environment_type();
 	$config = merge_config_settings( $config, $config['environments'][ $environment ] ?? [] );
 
 	return $config;
 }
 
+/**
+ * Merge settings in an existing configuration file.
+ *
+ * @param array $config Existing configuration.
+ *
+ */
+function validate_config_settings( array $config ) {
+	$environment = get_environment_type();
+	$error_type = $environment === 'local' ? E_USER_ERROR : E_USER_WARNING;
+	// Get configs to validate.
+	$configs = [
+		'default' => $config,
+	];
+	$registered_modules = Module::get_all();
+
+	// Validate all environment types locally to help avoid pushing unforeseen problems to production.
+	if ( $environment === 'local' ) {
+		$configs['local'] = merge_config_settings( $config, $config['environments']['local'] ?? [] );
+		$configs['development'] = merge_config_settings( $config, $config['environments']['development'] ?? [] );
+		$configs['staging'] = merge_config_settings( $config, $config['environments']['staging'] ?? [] );
+		$configs['production'] = merge_config_settings( $config, $config['environments']['production'] ?? [] );
+	} else {
+		$configs[ $environment ] = merge_config_settings( $config, $config['environments'][ $environment ] ?? [] );
+	}
+
+	foreach ( $configs as $type => $conf ) {
+		foreach ( $conf['modules'] as $module_name => $module ) {
+			if ( ! array_key_exists( $module_name, $registered_modules ) && ! array_key_exists( 'entrypoint', $module ) ) {
+				trigger_error(
+					sprintf(
+						'Custom modules should have entrypoint property! Your module %1$s is missing the entrypoint property at %2$s environment!',
+						$module_name,
+						$type
+					),
+					$error_type
+				);
+			}
+		}
+
+		foreach ( $conf as $key => $value ) {
+			if ( $key !== 'modules' && $key !== 'environments' ) {
```

Comment:
So here we want to check if the top level `$key` is a registered module name that we know rather than just checking if it's not `modules` or `environments`. We want to catch errors where a known module name is being used incorrectly at the top level rather than any arbitrary properties - we do want to allow those. Does that make sense?

So it'd be something like:

```suggestion
			if ( array_key_exists( $key, $registered_modules ) ) {
```